os/net/netdev/netdev_ioctl.c

```diff
@@ -1063,7 +1129,12 @@ int netdev_ioctl(int sockfd, int cmd, unsigned long arg)
 	}
 
 	/* Execute the command */
-	ret = netdev_ifrioctl(sock, cmd, (FAR struct ifreq *)((uintptr_t)arg));
+#ifdef CONFIG_NET_LWIP
```

Comment:
If all these configs are enabled and cmd is SIOCADDRT, it will call all this ioctl...
Why don't you make branch here??
Comparing variable is always cheaper than comparing return of function....

Reply:
I just followed the previous format. I will consider your comment.

Reply:
@Taejun-Kwon We discussed this and decided to keep current codes. Checking commands in each function would be better for a maintenance and the cost of this way is not that high.

Reply:
It is not good option, when code is running, switch case check all case sequentially.
So your code check all cases in all ioctl...
```
movf      SWITCH, w
xorlw     CASE1
btfsc     STATUS, Z                ; If SWITCH = CASE1, jump to LABEL1
goto      LABEL1
xorlw     CASE2^CASE1
btfsc     STATUS, Z                ; If SWITCH = CASE2, jump to LABEL2
goto      LABEL2
xorlw     CASE3^CASE2
btfsc     STATUS, Z                ; If SWITCH = CASE3, jump to LABEL3
goto      LABEL3
```
It is basic assembly of switch-case...
Of course choice is yours, because you are owner.